                                                                                          Created by Maria Dobko

# Part I: Hypothesis testing

In [55]:
n <- 5  #id-number

ak_vector <- c()

#generate ak values
for(k in 1:150) {
    ak <- k * log(k^2 * n + pi)
    x <- ak - floor(ak)
    ak_vector[k] <- x 
}

length(ak_vector)
# Generate X1,... X100 samples
x <- qnorm(ak_vector[1:100])
# Generate Y1,... X50 samples
y <- qnorm(ak_vector[101:150])
x
y

[1] 150

[1] -1.29891927 -0.57317504  0.31189423  0.47380456 -0.62593367 -0.63855109
  [7]  0.24816506 -0.75633309 -1.25616876 -0.81088357  0.03631677 -2.72036229
 [13]  0.41142279 -0.07390217 -0.18936279  0.03276124  0.60261627 -1.57059402
 [19]  0.00261240 -1.65030320  0.51876355 -0.14579039 -0.59145468 -0.84426945
 [25] -0.82438939 -0.55174781 -0.11739018  0.49626162  2.97096389 -0.31622223
 [31]  0.91553156 -0.44339532  1.28146400  0.07999563 -0.76331226  1.88934929
 [37]  0.75243744  0.33641204  0.10545948  0.00896810  0.03807450  0.19064353
 [43]  0.47792937  0.98722400 -1.77087666 -0.57451546  0.17846991  1.28959894
 [49] -0.60175663  0.48668992 -1.08091908  0.33768530 -0.98217859  0.61753356
 [55] -0.42023102  2.03612920  0.40382041 -0.33285939 -1.19060818  1.27076166
 [61]  0.56106540  0.15083651 -0.15214438 -0.38295170 -0.54326083 -0.62237560
 [67] -0.61121580 -0.51274372 -0.33975888 -0.10304618  0.20057188  0.60730865
 [73]  1.30852654 -1.23438564 -0.41399857  0.24563077  1.17956704 -0.87499217
 [79]  0.06599418  1.21150029 -0.60379308  0.47758567 -1.18273354  0.19546747
 [85] -1.55202750  0.16798165 -1.30273409  0.38250162 -0.75772606  0.92352755
 [91] -0.14543288 -1.37726338  0.66816998 -0.16806305 -1.08110712  1.11435769
 [97]  0.29468995 -0.29249931 -0.93680240  2.14720277

[1]  0.8919657017  0.4214336601  0.0812452811 -0.1996139372 -0.4436767486
 [6] -0.6574866150 -0.8377015193 -0.9722736713 -1.0436009827 -1.0383108460
[11] -0.9582301660 -0.8189649379 -0.6381398062 -0.4270274029 -0.1885750028
[16]  0.0831618880  0.4078459670  0.8436577356  1.7481712487 -1.0993687251
[21] -0.4440809135  0.0946668496  0.7170320052 -2.5624790233 -0.6335115297
[26]  0.0933923719  0.9426973417 -1.1136764898 -0.1156584374  0.8089180757
[31] -1.0675514606  0.0257506985  1.2417444954 -0.5521128617  0.5329401315
[36] -1.1249508934  0.1823701377 -1.8956680396  0.0003595567  2.1882414112
[41] -0.0363771190  3.2864099004  0.0691867140 -1.4783809148  0.3202636751
[46] -0.8582680187  0.7682405423 -0.3162947205  2.2105059981  0.2805281278

### Problem 1. H 0 : μ 1 = 0 vs. H 1 : μ < 0;σ1^ 2 is unknown.

Standard test:
The t test compares two averages (means) and tells if they are different from each other. The t test also tells how significant the differences are.
t-test T-test is used for the analysis of the two means. This test is good for this case since we have two independentsamples both sampled from a Normal distribution and we need to test their means when the variances are unknown.


In [56]:
t_test = t.test(x, y, alternative = "less", mu = 0, paired = FALSE, var.equal = FALSE, conf.level = 0.95)
t_test


	Welch Two Sample t-test

data:  x and y
t = -0.037062, df = 85.95, p-value = 0.4853
alternative hypothesis: true difference in means is less than 0
95 percent confidence interval:
      -Inf 0.2863109
sample estimates:
  mean of x   mean of y 
-0.07215216 -0.06562503 


In [57]:
print("Critical value: ")
alpha <- .05 
t_test.alpha = qt(1-alpha, df=149) 
t_test.alpha

[1] "Critical value: "


[1] 1.655145

In [58]:
t_test.half.alpha = qt(1-alpha/2, df=149)
c(-t_test.half.alpha, t_test.half.alpha) 

[1] -1.976013  1.976013

The test statistic -0.0370 lies between the critical values -1.9760, and 1.97601. Hence, at .05 significance level, we do not reject the null hypothesis.

In [59]:
t_test$p.value

[1] 0.4852608

If the P value is less than the chosen significance level -> reject the null hypothesis.
Significance level is 0.05, the corresponding confidence level is 95%.
P-value is 0.4853 and it's bigger than significance level, so H0 should be accepted!


### Problem 2. H 0 : μ 1 = μ 2 vs. H 1 : μ 1 != μ 2 ; σ 1 ^2 = σ 2^ 2 = 2

In [63]:
#t.test(x, y, var.equal=TRUE)
ttest <- t.test(x, y, var.equal=TRUE, conf.level=0.95)
ttest


	Two Sample t-test

data:  x and y
t = -0.039003, df = 148, p-value = 0.9689
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.3372307  0.3241764
sample estimates:
  mean of x   mean of y 
-0.07215216 -0.06562503 


H0 : u1 = u2 (H0: means are equal)
H1: )alternative) means are not equal

Standard test: two-sided t-test. Can be used for the cases with two independent population samples both sampled from Normal distribution and when the null hypothesis is that the two means are equal, and the alternative is that they are not.

95 percent confidence interval: (-0.3372307, 0.3241764).

Significance level 0.05: t = -0.039003, test statistics falls in the rejection region(outside the confidence interval). The null hypothesis should be rejected.

p-value = 0.9689 larger than the significance level, thus we accept H0.

### Problem 3. H 0 : σ 1^ 2 = 1 vs. H 1 : σ 1 ^2 ! = 1; μ 1 = 0

In [64]:
chi_test <- chisq.test(x[1:50], y)
chi_test

Warning message in chisq.test(x[1:50], y):
“Chi-squared approximation may be incorrect”


	Pearson's Chi-squared test

data:  x[1:50] and y
X-squared = 2450, df = 2401, p-value = 0.2382


We use chi-sqaured test when we observe n independent trials and note the numbers of
times P that each of k possible outcomes occurs, i.e., x = (x 1 , ..., x k ), sum(Xj) = n

A chi-squared test can be used to attempt rejection of the null hypothesis that the data are independent.

### Problem 4. H 0 : σ 1^ 2 = σ 2^ 2 vs. H 1 : σ 1 2 > σ 2^ 2 ; μ 1 and μ 2 are unknown

In [65]:
var.test(x_samples, y_samples, alternative = "greater")



	F test to compare two variances

data:  x_samples and y_samples
F = 0.73622, num df = 99, denom df = 49, p-value = 0.9002
alternative hypothesis: true ratio of variances is greater than 1
95 percent confidence interval:
 0.4808784       Inf
sample estimates:
ratio of variances 
         0.7362242 


H0: var1^2 = var2^2  
H1: var1^2 > var2^2
means are unknown

Standard test: one-sided f-test.
It is used for finding out whether there is any variance within the samples. F-test is the ratio of variance of two samples. It is run when we want to compare two variances, which is our case and the f-test is suitable in this problem. If F-test deviates a lot from 1, evidence become stonger that variances are unequal.

Rejection region:
95 percent confidence interval is (0.4808784, Inf).

Significance level 0.05: H0 should be accepted.

p-value = 0.9002 larger than the significance level, so we accept the null hypothesis.